src/functions.jl

```diff
@@ -514,25 +541,48 @@ for T in (:Taylor1, :TaylorN)
         @inline function asin!(c::$T{T}, a::$T{T}, r::$T{T}, k::Int) where {T<:Number}
             if k == 0
                 a0 = constant_term(a)
-                @inbounds c[0] = asin( a0 )
-                @inbounds r[0] = sqrt( 1 - a0^2 )
+                if $T == Taylor1
+                    @inbounds c[0] = asin( a0 )
+                    @inbounds r[0] = sqrt( 1 - a0^2 )
+                else
+                    @inbounds c[0][1] = asin( a0 )
+                    @inbounds r[0][1] = sqrt( 1 - a0^2 )
+                end
                 return nothing
             end
             zero!(c, k)
-            if $T == Taylor1
-                @inbounds c[k] = (k-1) * r[1] * c[k-1]
-            else
-                @inbounds mul!(c[k], (k-1) * r[1], c[k-1])
-            end
-            @inbounds for i in 2:k-1
+            @inbounds for i in 1:k-1
                 if $T == Taylor1
                     c[k] += (k-i) * r[i] * c[k-i]
                 else
-                    mul!(c[k], (k-i) * r[i], c[k-i])
+                    mul!(c[k], r[i], c[k-i], scalar=k-i)
+                end
+            end
+            # sqrt!(r, 1-a^2, k)
+            # Compute auxiliary term s=1-a^2
+            s = ($T)(zero(a[0]), a.order)
+            one_s = one(s)
+            for i = 0:k
+                sqr!(s, a, i)
+                subst!(s, s, i)
+                if i == 0
+                    if $T == Taylor1
+                        s[0] = one(s[0]) - s[0]
+                    else
+                        s[0][1] = one(s[0][1]) - s[0][1]
+                    end
+                    add!(s, one_s, s, 0)
+                end
+            end
```

Comment:
Same comment about order of `if` and `for`

Reply:
Done 👍

Reply:
(actually the code changed here a bit so this `for` is not there anymore)